graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/DgsGraphQLConfigurationProperties.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+import org.springframework.boot.context.properties.bind.DefaultValue
+import org.springframework.validation.annotation.Validated
+
+/**
+ * Configuration properties for DGS.
+ */
+@ConstructorBinding
+@ConfigurationProperties(prefix = "dgs.graphql")
+@Validated
+@Suppress("ConfigurationProperties")
+data class DgsGraphQLConfigurationProperties(
```

Comment:
I'm wondering how the set of properties will grow over time, and if those properties are "common" or specific to certain technologies. E.g. I might imagine spring-data integration, which would have its own set of configurations. That configuration probably doesn't belong in a common module. That would speak for the fine-grained approach.

For the coarse-grained approach, does it _need_ to live in the dgs core module? Could we instead move it either to a separate module? That way it can still be all together but without the need to "bloat" the core module?

Reply:
I think the fine grained approach will work out well. I'll pivot and have something to look out in a bit.